framework/src/governance/lib/datazone/datazone-msk-environment-authorizer.ts

```diff
@@ -0,0 +1,117 @@
+ * The workflow is triggered by an event sent by the central authorizer construct.
+ * It creates IAM policies required for the Kafka client to access the relevant topics.
+ * It supports MSK provisioned and serverless, in single and cross accounts, and grant/revoke requests.
+ *
+ * @example
+ * new dsf.governance.DataZoneMskEnvironmentAuthorizer(this, 'MskAuthorizer', {
+ *   domainId: 'aba_dc999t9ime9sss',
+ * });
+ */
+export class DataZoneMskEnvironmentAuthorizer extends TrackedConstruct {
+
+  /**
+   * The IAM role used to grant access to Kafka topics
+   */
+  public readonly grantRole: IRole;
+  /**
+   * The lambda function used to grant access to Kafka topics
+   */
+  public readonly grantFunction: IFunction;
+  /**
+   * The IAM role used by the environment authorizer State Machine
+   */
+  public readonly stateMachineRole: IRole;
+  /**
+   * The environment authorizer State Machine
+   */
+  public readonly stateMachine: IStateMachine;
+
+  private readonly removalPolicy: RemovalPolicy;
+
+  /**
+   * Create an instance of the DataZoneMskEnvironmentAuthorizer construct
+   * @param scope The CDK Construct scope
+   * @param id The CDK Construct id
+   * @param props The props for the DataZoneMskEnvironmentAuthorizer construct
+   */
+  constructor(scope: Construct, id: string, props: DataZoneMskEnvironmentAuthorizerProps) {
+    const trackedConstructProps: TrackedConstructProps = {
+      trackingTag: DataZoneMskEnvironmentAuthorizer.name,
+    };
+
+    super(scope, id, trackedConstructProps);
+
+    this.removalPolicy = Context.revertRemovalPolicy(this, props.removalPolicy);
+
+    const grantRole = new Role(this, 'GrantRole', {
+      roleName: `${DataZoneMskCentralAuthorizer.AUTHORIZER_NAME}GrantFunction`,
+      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+      managedPolicies: [
+        ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
+      ],
+      inlinePolicies: {
+        IamPermissions: new PolicyDocument({
+          statements: [
+            new PolicyStatement({
+              effect: Effect.ALLOW,
+              actions: [
+                'iam:PutRolePolicy',
+                'iam:DeleteRolePolicy',
+              ],
+              resources: ['*'],
+            }),
+            new PolicyStatement({
+              effect: Effect.ALLOW,
+              actions: [
+                'kafka:GetClusterPolicy',
+                'kafka:PutClusterPolicy',
+                'kafka:DeleteClusterPolicy',
+              ],
+              resources: ['*'],
+            }),
+          ],
+        }),
+      },
+    });
+
+    this.grantRole = grantRole;
+
+    this.grantFunction = new Function(this, 'GrantFunction', {
+      runtime: Runtime.NODEJS_20_X,
+      handler: 'index.handler',
+      code: Code.fromAsset(__dirname + '/resources/datazone-msk-authorizer-grant/'),
+      role: this.grantRole,
+      timeout: Duration.seconds(60),
+      environment: {
```

Comment:
encrypt the environment variablem through a CMK provided by the user, else no encryption which default to `KMS`